packages/rest/src/sequence.ts

```diff
@@ -121,3 +122,83 @@ export class DefaultSequence implements SequenceHandler {
     }
   }
 }
+
+/**
+ * A sequence implementation using middleware chains
+ */
+export class MiddlewareSequence implements SequenceHandler {
+  static defaultOptions: InvokeMiddlewareOptions = {
+    chain: RestTags.REST_MIDDLEWARE_CHAIN,
+    orderedGroups: [
+      // Please note that middleware is cascading. The `SendResponse` is
+      // added first to invoke downstream middleware to get the result or
+      // catch errors so that it can produce the http response.
+      'sendResponse',
+
+      // default
+      'cors',
+      'apiSpec',
+
+      // default
+      'middleware',
+
+      // rest
+      'findRoute',
+
+      // authentication
+      'authentication',
```

Comment:
For this to work, user has to create a `Provider<Middleware>` and set its `group` value to `authentication`; and many such authentication middleware can be added and anyone of them can cause the authentication to fail, right?

Reply:
A group does not need to have middleware. It's only for ordering purpose.